src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodBeAnalysis.Remote.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Razor.LanguageClient.Cohost;
+using Microsoft.VisualStudio.Razor.Settings;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostOnAutoInsertEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Theory]
+    [InlineData("PageTitle", "$0</PageTitle>", ">")]
+    [InlineData("div", "$0</div>", ">")]
+    [InlineData("text", "$0</text>", ">")]
+
+    public async Task Component_AutoInsertEndTag(string startTag, string endTag, string triggerCharacter)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}$$
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, endTag, triggerCharacter);
+    }
+
+    [Theory]
+    [InlineData("div style", "\"\"", "=")]
+    public async Task Component_AutoInsertAttributeQuotes(string startTag, string insertedText, string triggerCharacter)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}$$
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, insertedText, triggerCharacter, createDelegatedResponse: true);
+    }
+
+    [Theory]
+    [InlineData("""
+            @code {
+                //$$
+                void TestMethod() {}
+            }
+            """,
+        "/// <summary>", "/")]
```

Comment:
IMO this is much harder to read than just having the test input in the test.

Reply:
I am planning to add more cases here, otherwise I agree.

Reply:
If you're going to add more cases, please just add them as separate tests, rather than separate theories. This is really really hard to read. Having separate tests means you can also given them unique names so they describe what they're testing, and the only code that would be duplicated is the single line of code that runs the test. I see no benefit to theories like this, and only downside.

Reply:
Sure, I can see that.